Providers/Profile.swift

```diff
@@ -603,10 +621,37 @@ public class BrowserProfile: Profile {
         private var syncReducer: AsyncReducer<EngineResults, EngineTasks>?
 
         private func beginSyncing() {
+            syncState = .InProgress
             notifySyncing(NotificationProfileDidStartSyncing)
         }
 
         private func endSyncing() {
+            // loop through status's and fill sync state
+            if let syncResult = syncReducer?.terminal.value {
```

Comment:
Looks like `endSyncing` is called after `syncReducer.terminal` completes [1] but since we're using the `>>>` operator, the result is ignored. It also means that `endSyncing` only gets called if `syncReducer.terminal` was successful [2]. Instead of modifying `endSyncing`, why not add the flip to `onFinishSyncing`?

[1] https://github.com/mozilla/firefox-ios/pull/1796/files#diff-7a45ce0d70d55bdf297fcacd47e4e874R1032
[2] https://github.com/mozilla/firefox-ios/blob/master/Utils/DeferredUtils.swift#L27
[3] https://github.com/mozilla/firefox-ios/blob/master/Providers/Profile.swift#L785